frontend/src/Components/RoadDetails/RoadImage.tsx

```diff
@@ -56,6 +100,74 @@ const RoadImage: React.FC<Props> = ({ type, id, selectedType }) => {
     setHasFiltered(true);
   }, [allImages, selectedType]);
 
+  useEffect(() => {
+    if (containerRef.current) {
+      const container = containerRef.current as HTMLElement;
+      const observer = new ResizeObserver(() => {
+        // Handle size changes
+        handleScroll();
+      });
+
+      // Start observing the container
+      observer.observe(container);
```

Comment:
Can you put the addEventListener and observer at the place? So that the event are declared at the same place, i think it make more sense when you read the code

Reply:
what do you mean by "at the place"? everything at the end of the use effect?